BraveWallet/Crypto/Stores/CryptoStore.swift

```diff
@@ -37,6 +37,12 @@ public class CryptoStore: ObservableObject {
   private let txService: BraveWalletTxService
   private let ethTxManagerProxy: BraveWalletEthTxManagerProxy
   
+  private var unapprovedTransactions: [BraveWallet.TransactionInfo] = [] {
+    didSet {
+      hasUnapprovedTransactions = !unapprovedTransactions.isEmpty
+    }
+  }
```

Comment:
Remove this entirely, set `hasUnapprovedTransactions` based on where `unapprovedTransactions` was being set (observer probably?)